odml/doc.py

```diff
@@ -25,10 +25,32 @@ class BaseDocument(base.sectionable, Document):
 
     def __init__(self, author=None, date=None, version=None, repository=None):
         super(BaseDocument, self).__init__()
-        self._author = author
+        self._author = self.escape_ignore(author)
         self._date = date # date must be a datetime
-        self._version = version
-        self._repository = repository
+        self._version = self.escape_ignore(version)
+        self._repository = self.escape_ignore(repository)
+
+    def escape_ignore(self, value):
+        if value!=None:
+            escape_dict={'\a':r'\a',
+                         '\b':r'\b',
+                         '\c':r'\c',
+                         '\f':r'\f',
+                         '\n':r'\n',
+                         '\r':r'\r',
+                         '\t':r'\t',
+                         '\v':r'\v',
+                         '\'':r'\'',
+                         '\"':r'\"'}
+            new_value = ''
+            for char in value:
+                try: 
+                    new_value += escape_dict[char]
+                except KeyError: 
+                    new_value += char
+            return new_value
+        else:
+            return None
```

Comment:
The `escape_ignore` method has a few issues.
1. Comparisons with `None` should be done using `is` and `is not` instead of `==` and `!=`.
2. The character-by-character iteration and appending is hugely inefficient. For one, it would be much cleaner to just use `str.replace()`, even though that's not great either.
3. Exception handling for each character compounds the problem even further. Escape characters are rare in the strings we're cleaning, so the `except` block will be run almost always while the `try` will run through very rarely. Having a non-escape character isn't _exceptional_.

Overall, though, are you sure this solves the issue in a general manner? Could you include some tests for this to check if we're getting the correct behaviour?